### PR TITLE
Added Header & UserToken components and collapsible feature

### PR DESCRIPTION
**Pull request by tcelovsky**
Co-authored-by: Yolanda Haynes <yhaynes@giyo.us>

## Description

Created Header and UserToken component to display the app’s name and the token throughout the app. The token will only display once the user is signed in. Added new dependency in order for the app to use the collapsible feature.

## Related Issue

#13 

## Acceptance Criteria

- App has a header that is displayed throughout the project
- UserToken is displayed throughout the app to allow the user to share the shopping list with others
- List items can collapse/expand to display a more succinct list

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
| ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

- From your terminal, pull down this branch with `git pull origin tc-yh-add-header-user-token` and check that branch out with `git checkout tc-yh-add-header-user-token`.
- Then `npm install` or `yarn to install` the newly added dependency locally and `npm start` or `yarn start` to launch the app.
- Ensure that the app name "TCL-27" is displayed throughout the app.
- Ensure that the current user token is displayed on all pages of the app once the user is logged in and is not displayed if the user is not logged in.
- Ensure that the list items can expand and collapse by clicking on the name of the items in the List view.

### CODE REVIEW

**Review by AirZT11 — APPROVED** [review]
Nicely done! Love the collapsible feature for the cards! Looks slick! Also, great job on displaying the new header and token component.

File: src/components/GroceryCard.js [hunk 1]
```diff
@@ -158,17 +159,18 @@ const GroceryCard = ({ item }) => {
...
+          LAST PURCHASE DATE:{' '}
+          {item.lastPurchase ? item.lastPurchase.toDate().toDateString() : 'NA'}
+        </p>
+        <p>
+          NEXT ESTIMATED PURCHASE DATE:{' '}
+          {item.nextPurchaseDate
+            ? item.nextPurchaseDate.toDate().toDateString()
+            : ''}
+        </p>
+      </Collapsible>
```
Comment (Dani-Kirby) [inline src/components/GroceryCard.js:173]: Love this collapsible component! During testing noticed that it is difficult to click the specific spot we need to expand. Would be worthwhile to play with this and maybe add something that indicates the item is expandable? I'm thinking something like an ellipsis could work here.

File: src/components/Header.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const Header = () => {
+  return (
+    <div>
+      <h1>TCL-27</h1>
```
Comment (Dani-Kirby) [inline src/components/Header.js]: Woot woot!

**Review by Dani-Kirby — APPROVED** [review]
This is looking good! We'll want to flesh out the expandable field behavior, but beyond that I'm pleased with the functionality here. Also appreciate the new components making the 'single source of truth' a bit easier to maintain.

File: src/components/GroceryCard.js (hunk 1 above)
Comment (tcelovsky) [reply src/components/GroceryCard.js:173]: Yeah, definitely hard to know what to click. We thought that maybe margins/padding would help (and good point on adding an ellipsis!) once CSS is implemented, but didn't want to start adding styles until we knew which way we were gonna go.

File: src/components/Header.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const Header = () => {
+  return (
+    <div>
+      <h1>Woot Woot!</h1>
```
Comment (grellyd) [inline src/components/Header.js:6]: Heck yeah~

**Review by grellyd — APPROVED** [review]
Looking great! 

Just one thing to keep an eye out for when you do the styling: the padding on the list is uneven: 
padding
padding2

**Pull request merged**